### ISSUE TITLE
Create Project || Project creation unsuccessful when 100+ chars are entered

### ISSUE THREAD

**Issue by p-tal** [Code4SocialGood/c4sg-web#766]
Steps to reproduce:
1. Login with nonprofit user: testnonprofituser2@gmail.com / Opensource5social!
2.Goto My settings-> Create Project
3.Enter details with project name in long string (more than 120 chars).
4.Enter project desc.
5.Save project.
Actual Result: The project name and description is blank and creation is unsuccessful. 
Please refer to below mentioned project:
http://dev.code4socialgood.org/project/view/252
![long stringproject](https://user-images.githubusercontent.com/29553195/28193866-25bf2460-680e-11e7-94e8-98dccc8fcddd.jpg)

**Comment by Code4SocialGood** [comment]
**Background**
1. The Project Name at backend database columns is defined as char(100). If frontend sends in a value longer than 100 char, backend will throw database exception: Caused by: com.mysql.jdbc.MysqlDataTruncation: Data truncation: Data too long for column 'name' at row 1.
2. On backend, we created an issue to let Restful API to indicate the max length the field could accept.
3. On frontend, please validate the field length. 

**What to fix on frontend**
1. Validate the max length of every input field.
- user-edit.component.html
Username: 100
First Name: 100
Last Name: 100
State: 100
Title: 100
Enter your skill: 100
LinkedIn URL: 100
Github URL: 100
My Website: 100
Slack Username: 25
Phone: 16

- project-edit.component.html
Project Name: 100
City: 100
State: 100
Enter your skill: 100

- organization-edit.component.html
Organization Name: 100
Website: 100
EIN: 20
Address 1: 100
Address 2: 100
City: 100
State: 100
Zip Code: 16

2. Present the validation error like below:
http://materializecss.com/forms.html
![image](https://user-images.githubusercontent.com/24813620/28194519-73bde556-680a-11e7-81e6-a736ced0b116.png)
http://demo.geekslabs.com/materialize-v1.0/form-validation.html

3. Let's leave the text area validation (description text area that is max of 1000 char) untouched. In this issue we only change the input field validation.

**Comment by webdev-burd** [comment]
1. I created a public variable called maxCharacters on *.component.ts.

```javascript
export class UserEditComponent implements ... {
  ...
  public maxCharacters: number = 100;
 ...
}
```
and then used the validation directive `[maxlength]="maxCharacters"` on all inputs listed above.
I'm working on 2 and 3 now.

**Comment by Code4SocialGood** [comment]
Resolved in PR #904
